### PR TITLE
fix mime checkes

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #16086

### Summary of Changes

If you have installed 3.7.1 and try to upload using com_media a file which is not a image you got an error that you can't upload it.

I have now:
- [X] Improved the checks
- [X] show the failing mime to the use
- [X] make sure that images are checked with the image functions and other files with the other checks
- [X] implemented a fallback for images
- [X] fixed a longstanding issuse with finfo_open (FILEINFO_MIME vs. FILEINFO_MIME_TYPE)

### Testing Instructions

- install 3.7.1
- try to upload a PDF file.
- it fails
- try to upload a jpg file
- it works
- apply this patch
- try the PDF and the image again
- it works now

Now go to the options and don't allow uploading jpg images and PDF files (by removing the jpg & PDF mime from the allowed mime types

- try to upload a jpg again
- it should not work but now show the failing mime type. (image/jpg)
- do the same for a PDF
- it should also show the failing mime.

### Expected result

If allowed the upload should work if not it should show the mime we detected to the user.

### Actual result

non image files can't be uploaded using the media manger

### Documentation Changes Required

none.

### CODE REVIEW

File: libraries/cms/helper/media.php [hunk 1]
```diff
@@ -199,10 +213,24 @@ public function canUpload($file, $component = 'com_media')
...
+
+						// If the mime type is not allowed we don't upload it and show the mime code error to the user
+						if ($result === false)
+						{
+							$app->enqueueMessage(JText::_('JLIB_MEDIA_ERROR_WARNINVALID_MIMETYPE', $mime), 'error');
+
+							return false;
+						}
+					}
+					// We can't detect the mime type so it looks like a invalid image'
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `a` to `an` and remove `'`

File: libraries/cms/helper/media.php [hunk 2]
```diff
@@ -218,10 +246,24 @@ public function canUpload($file, $component = 'com_media')
...
-				if ($result === false)
+					// If the mime type is not allowed we don't upload it and show the mime code error to the user
+					if ($result === false)
+					{
+						$app->enqueueMessage(JText::_('JLIB_MEDIA_ERROR_WARNINVALID_MIMETYPE', $mime), 'error');
+
+						return false;
+					}
+				}
+				// We can't detect the mime type so it looks like a invalid image'
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `a` to `an` and remove `'`.

File: libraries/cms/helper/media.php [hunk 3]
```diff
@@ -47,60 +47,74 @@ public static function getTypeIcon($fileName)
...
+				finfo_close($finfo);
+			}
+		}
+		catch (Exception $e)
+		{
+			// If we have any kind of error here => false;
+			return false;
+		}
+
+		// If we cant detect the mime try it again
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `cant` to `can't`.

File: libraries/cms/helper/media.php [hunk 4]
```diff
@@ -199,10 +213,24 @@ public function canUpload($file, $component = 'com_media')
 				// If tmp_name is empty, then the file was bigger than the PHP limit
 				if (!empty($file['tmp_name']))
 				{
-					$result = $this->checkMimeType($file['tmp_name'], $component, true);
+					// Get the mime type this is a image file
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `a` to `an`. Add `;` after `type`.

**Comment by zero-24** [comment]
Should be fixed now thanks @Quy !

**Comment by PhilETaylor** [comment]
@rdeutz Joomla 3.7.2 please

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on efc334f069996e1af9683a7dd764f2ab27b73281<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by Quy** [comment]
Uploaded a PDF file and got the error `Invalid mime type detected.` and in the PHP error log:
```
PHP Notice:  Undefined variable: mime in C:\xampp\htdocs\joomla371\libraries\cms\helper\media.php on line 92
PHP Notice:  Undefined variable: mime in C:\xampp\htdocs\joomla371\libraries\cms\helper\media.php on line 98
```

File: libraries/cms/helper/media.php [hunk 5]
```diff
@@ -47,54 +47,73 @@ public static function getTypeIcon($fileName)
...
+				// We have mime magic.
+				$mime = mime_content_type($file);
+			}
+			elseif (function_exists('finfo_open'))
+			{
+				// We have fileinfo
+				$finfo = finfo_open(FILEINFO_MIME_TYPE);
+				$mime  = finfo_file($finfo, $file);
+				finfo_close($finfo);
+			}
```
Comment (wilsonge) [inline libraries/cms/helper/media.php]: Looks like from the test we have cases where none of these things exist!? Probably having a final else statement here with return false to reject everything if we can't detect things is going to be important

**Comment by N6REJ** [comment]
following

File: libraries/cms/helper/media.php [hunk 6]
```diff
@@ -47,60 +47,74 @@ public static function getTypeIcon($fileName)
...
+		try
+		{
+			if ($isImage && function_exists('exif_imagetype'))
+			{
+				$mime = image_type_to_mime_type(exif_imagetype($file));
+			}
+			elseif ($isImage && function_exists('getimagesize'))
+			{
+				$imagesize = getimagesize($file);
+				$mime      = (isset($imagesize['mime'])) ? $imagesize['mime'] : false;
```
Comment (Quy) [inline libraries/cms/helper/media.php]: c/s remove outside `()` since not required.

File: libraries/cms/helper/media.php [hunk 7]
```diff
@@ -49,45 +49,46 @@ public static function getTypeIcon($fileName)
...
-				elseif (function_exists('finfo_open'))
+				elseif ($isImage && function_exists('getimagesize'))
 				{
-					// We have fileinfo
-					$finfo = finfo_open(FILEINFO_MIME);
-					$mime  = finfo_file($finfo, $file);
-
-					finfo_close($finfo);
+					$imagesize = getimagesize($file);
+					$mime      = (isset($imagesize['mime'])) ? $imagesize['mime'] : false;
```
Comment (Quy) [inline libraries/cms/helper/media.php]: c/s remove outside `()` since not required.

**Comment by infograf768** [comment]
Except for the cs https://github.com/joomla/joomla-cms/pull/16091#discussion_r117162329 this works for me

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on efc334f069996e1af9683a7dd764f2ab27b73281<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
set RTC, @infograf768?

**Comment by infograf768** [comment]
Waiting for
`$mime      = (isset($imagesize['mime'])) ? $imagesize['mime'] : false;`
to be corrected to
`$mime      = isset($imagesize['mime']) ? $imagesize['mime'] : false;`

**Comment by wilsonge** [comment]
Quy's php notices need to be investigated before this can go RTC - clearly in some cases this isn't working fully

File: libraries/cms/helper/media.php (hunk 5 above)
Comment (zero-24) [inline libraries/cms/helper/media.php]: Fixed with: https://github.com/joomla/joomla-cms/pull/16091/commits/298959ec803d1aa698603f141430d1eb8358025a

**Comment by zero-24** [comment]
@Quy 

This should be fixed with: https://github.com/joomla/joomla-cms/pull/16091/commits/298959ec803d1aa698603f141430d1eb8358025a

But this shows that you have both checker not enabled which is a problem.

File: libraries/cms/helper/media.php (hunk 7 above)
Comment (zero-24) [inline libraries/cms/helper/media.php]: https://github.com/joomla/joomla-cms/pull/16091/commits/672e1836662663eb0473a05ce346202c2311e559

**Comment by zero-24** [comment]
All issues mentiond above should be fixed now please test.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 672e1836662663eb0473a05ce346202c2311e559<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

File: libraries/cms/helper/media.php [hunk 8]
```diff
@@ -47,54 +47,76 @@ public static function getTypeIcon($fileName)
...
+			}
+		}
+		catch (Exception $e)
+		{
+			// If we have any kind of error here => false;
+			return false;
+		}
+
+		// If we can't detect the mime try it again
+		if ($mime == 'application/octet-stream' && $isImage === true)
```
Comment (wilsonge) [inline libraries/cms/helper/media.php]: We should probably use `===` here for the mime check as now we are checking false to a string

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 672e1836662663eb0473a05ce346202c2311e559<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by wilsonge** [comment]
@Quy can you give this a test again please?

**Comment by franz-wohlkoenig** [comment]
@wilsonge can you give a Go if RTC-ready?

**Comment by wilsonge** [comment]
If this fixes the issues @Quy had then it's RTC ready :)

File: libraries/cms/helper/media.php (hunk 8 above)
Comment (zero-24) [inline libraries/cms/helper/media.php]: https://github.com/joomla/joomla-cms/pull/16091/commits/be99e2aed77d51b8ca8eee03aca9d9bed6ee65f3 done please retest

File: libraries/cms/helper/media.php [hunk 9]
```diff
@@ -88,14 +88,19 @@ private function checkMimeType($file, $component = 'com_media', $isImage = false
...
-
 			}
 			catch (Exception $e)
 			{
 				// If we have any kind of error here => false;
 				return false;
 			}
 
+			// If we cant detect the mime try it again
+			if ($mime == 'application/octet-stream' && $isImage === true)
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Earlier `$isImage` is tested without `--- true`. Do you want it to be consistent here?

File: libraries/cms/helper/media.php (hunk 4 above)
Comment (Quy) [inline libraries/cms/helper/media.php]: Change.

File: libraries/cms/helper/media.php (hunk 1 above)
Comment (Quy) [inline libraries/cms/helper/media.php]: Change

File: libraries/cms/helper/media.php [hunk 10]
```diff
@@ -218,10 +246,24 @@ public function canUpload($file, $component = 'com_media')
 			}
 			elseif (!in_array($filetype, $ignored))
 			{
-				$result = $this->checkMimeType($file['tmp_name'], $component, false);
+				// Get the mime type this is not a image file
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `a` to `an`. Add `;` after `type`.

File: libraries/cms/helper/media.php [hunk 11]
```diff
@@ -218,10 +246,24 @@ public function canUpload($file, $component = 'com_media')
...
+				$mime = $this->getMimeType($file['tmp_name'], false);
+
+				// Did we got anything usefull?
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change to `// Did we get anything useful?`

File: libraries/cms/helper/media.php [hunk 12]
```diff
@@ -199,10 +213,24 @@ public function canUpload($file, $component = 'com_media')
 				// If tmp_name is empty, then the file was bigger than the PHP limit
 				if (!empty($file['tmp_name']))
 				{
-					$result = $this->checkMimeType($file['tmp_name'], $component, true);
+					// Get the mime type this is a image file
+					$mime = $this->getMimeType($file['tmp_name'], true);
 
-					// If the mime type is not allowed we don't upload it
-					if ($result === false)
+					// Did we got anything usefull?
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change to `// Did we get anything useful?`

**Comment by franz-wohlkoenig** [comment]
@Quy can you give a Go for RTC if issues are fixed?

**Comment by Quy** [comment]
My testing machine does not have functions `mime_content_type` and `finfo_open`, therefore, no mime type is determined (false). The PHP notices are fixed. I will test it on a live server.

File: libraries/cms/helper/media.php [hunk 13]
```diff
@@ -58,6 +58,9 @@ public static function getTypeIcon($fileName)
 	 */
 	private function getMimeType($file, $isImage = false)
 	{
+		// If we can't detect anything mime is false
+		$mime = false;
+
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Remove whitespace.

**Comment by Quy** [comment]
> But this shows that you have both checker not enabled which is a problem.
Yes, this is the case with my XAMPP install, but not on the live server. This PR should be fine.

**Comment by Quy** [comment]
Don't RTC yet as there are minor changes to be made.

File: libraries/cms/helper/media.php [hunk 14]
```diff
@@ -49,45 +49,46 @@ public static function getTypeIcon($fileName)
 	/**
 	 * Checks the Mime type
 	 *
-	 * @param   string  $file       The filename or tmp_name
-	 * @param   string  $component  The optional name for the component storing the parameters
+	 * @param   string   $file       The filename or tmp_name
+	 * @param   string   $component  The optional name for the component storing the parameters
+	 * @param   boolean  $image      True if the passed file is a image else false
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change `a` to `an`.

**Comment by infograf768** [comment]
We still have some typos in comments

**Comment by brianteeman** [comment]
I spotted one typo and fixed it at https://github.com/joomla/joomla-cms/pull/16091/commits/0c89e50405576cb530845f45cc44d88e52804c18 i didn't immediately see any others

**Comment by zero-24** [comment]
Thanks for the corrections.

**Comment by infograf768** [comment]
RTC as the last changes only concerned typos in comments.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

File: libraries/cms/helper/media.php (hunk 9 above)
Comment (Quy) [inline libraries/cms/helper/media.php]: Consistency wise...ok as it???

**Comment by zero-24** [comment]
Thanks. I'm very sorry that I broke that with my first patch.

**Comment by EndeavorC** [comment]
Would someone please confirm that the changes in the media.php file correct the issue. I performed the code changes as stated in the fix and it did not resolve this issue for my site. For the site I support the users have editor (front end) access only, trying to upload an image from the front end fails with the invalid mime type error. I did not specify the language files but they were updated as well.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by zero-24** [comment]
Hmm. What kind of error do you get if any mime could be detected we show it now to you. (after the changes) can you please confirm that you have atleast on of the modules installed? Please also share the image you tried to upload so we can reproduce and test. Thanks

**Comment by EndeavorC** [comment]
I receive the following errors when logged on with a user that has editor permissions with any jpg file, I have not tried other file types:

Error
Illegal mime type detected: image/jpeg

Warning
Unable to Upload File.

I am able to upload jpg files from the administrator (back end) of the website. I have attached the file with the PHP information for the server.

Unfortunately for my site this bug is a show stopper, I had to upload the 3.7.0 files to replace the affected files for our volunteers to work. 

Please let me know if you need any other information.

Thank you

[systeminfo-2017-05-20.txt](https://issues.joomla.org/uploads/1/871e2362bd4c05204672ea1db145601c.txt)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by zero-24** [comment]
> I receive the following errors when logged on with a user that has editor permissions with any jpg file, > I have not tried other file types:
>
> Error
> Illegal mime type detected: image/jpeg


Please check the `Legal MIME Types` setting from the com_media options in the backend. And post what is is this field.

System -> Global Configuration -> Media -> Legal MIME Types.

> I am able to upload jpg files from the administrator (back end) of the website. I have attached the file with the PHP information for the server.

hmm that sounds not correct but it looks like the file is missing here.

**Comment by EndeavorC** [comment]
image/jpeg,image/gif,image/png,image/bmp,application/x-shockwave-flash,application/msword,application/excel,application/pdf,application/powerpoint,text/plain,application/x-zip<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by zero-24** [comment]
>  I performed the code changes as stated in the fix and it did not resolve this issue for my site.

Just to be sure how did you apply the changes? I hope by a complete file replace for the media file? As I can't find a way to get your result by your configuration.

Please try to replace your media file with this: https://raw.githubusercontent.com/joomla/joomla-cms/staging/libraries/cms/helper/media.php

**Comment by EndeavorC** [comment]
I went to the GitHub link; view the entire file contents for each file, copy/paste into a text editor (textwrangler), save the file as the same name, rename the existing file then upload the new file.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by wojsmol** [comment]
@EndeavorC Have you used raw view?

**Comment by zero-24** [comment]
@EndeavorC ok i have finaly found something that could be the issue.

Please run the following SQL
```sql 
select * from `#__extensions` where `element` = `com_media`;
```

And post the result here.

**Comment by EndeavorC** [comment]
Hello, 

When I run the above SQL query in myPHPAdmin I get the following:

Error
SQL query: Documentation

select * from `#__extensions` where `element` = `com_media` LIMIT 0, 25


MySQL said: Documentation

#1146 - Table '<database>.#__extensions' doesn't exist

Sorry this is not much help<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by zero-24** [comment]
Yes you need to replace #__ with your database prefix

**Comment by EndeavorC** [comment]
Error
SQL query: Documentation


select * from `<prefix>_extensions` where `element` = `com_media` LIMIT 0, 25
MySQL said: Documentation

#1054 - Unknown column 'com_media' in 'where clause'<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by EndeavorC** [comment]
If I view the com_media table I have the following, not sure if this is what you are looking for:

name: com_media

type: component

element: com_media

manifest_cache:

{"name":"com_media","type":"component","creationDate":"April 2006","author":"Joomla! Project","copyright":"(C) 2005 - 2017 Open Source Matters. All rights reserved.","authorEmail":"admin@joomla.org","authorUrl":"www.joomla.org","version":"3.0.0","description":"COM_MEDIA_XML_DESCRIPTION","group":"","filename":"media"}

params:

{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,swf,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,SWF,TXT,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","check_mime":"1","image_extensions":"bmp,gif,jpg,png","ignore_extensions":"","upload_mime":"image\/jpeg,image\/gif,image\/png,image\/bmp,application\/x-shockwave-flash,application\/msword,application\/excel,application\/pdf,application\/powerpoint,text\/plain,application\/x-zip","upload_mime_illegal":"text\/html"}<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16091">issues.joomla.org/tracker/joomla-cms/16091</a>.</sub>

**Comment by zero-24** [comment]
This is what I was looking for thanks. Can you please go to the com_media options and save them again. And than try uploading a image from the frontend again?

If this still don't work please send me a backup + the file you try to upload to tobias.zulauf[at]community.joomla.org as i'm running out of ideas what the issue could be. And I can not reproduce the problems.

**Comment by EndeavorC** [comment]
Saving the options did not solve the issue, what type of backup are you asking for?

**Comment by zero-24** [comment]
file and database you can clear the content table if needed. So I can debug the issue.

**Comment by gwsdesk** [comment]
If I understand all properly I can mention that I have tested this item ✅ successfully on [672e183](https://github.com/joomla/joomla-cms/commit/672e1836662663eb0473a05ce346202c2311e559)

**Comment by EndeavorC** [comment]
What permissions are you testing with for the user account? I am using an account with publisher. If I replace the media.php file with the one from 3.7.0 it works but not the updated one here or the one that was installed with 3.7.1.

**Comment by sedonasky** [comment]
I updated to 3.7.2 but still cannot upload PDFs. I cleared the cache, etc. and tried to upload with Super Admin permissions. I still get this error (so clients are still very unhappy).

Error
Invalid mime type detected.

**Comment by zero-24** [comment]
@sedonasky 

> Error
> Invalid mime type detected.

this means that `finfo_open` and `mime_content_type` did not work or are not available. Can you please check that they are installed and enabled on your server?

**Comment by gwsdesk** [comment]
I cannot replicate this on our servers https://gws-host.com/systems-technology

Imho this is a server issue/settings and not related to Joomla

**Comment by Quy** [comment]
> this means that finfo_open and mime_content_type did not work or are not available. Can you please check that they are installed and enabled on your server?

Maybe include in the error message that `finfo_open` and `mime_content_type` could not be found and to contact their hosting provider.

**Comment by gwsdesk** [comment]
@Quy which would mean that for every specific hosting config with disabled functions we would have to create exemptions/messages? Does that make sense?

**Comment by jen4web** [comment]
I am having the same problem. Site is hosted with Rochen, PHP 7.0.19. Site is upgraded to 3.7.2, and I am uploading a PDF (131K) as a super user.

**Comment by zero-24** [comment]
Yes please check that one or both php functions are enabled.

**Comment by fperillo** [comment]
I confirm this patch is not working on 3.7.1 and 3.7.2, due to mime_content_type
<?php echo mime_content_type('scheda2017.pdf'); ?>
PHP Fatal error:  Call to undefined function mime_content_type() in - on line 1

OpenSuse 12.3 has php5-fileinfo extension module that is not installed on the server...

**Comment by zero-24** [comment]
I hope The mime type fatal error is not shown to the user?

But yes if both methods are not there there is no upload allowed as we cant check the mime types which is required for Security reasons.

**Comment by fperillo** [comment]
I understand, and I really don't know Joomla developers guidelines and, believe me, I don't want to be rude, but after a hot phone call with one user... 
according to
https://downloads.joomla.org/technical-requirements
there is no need to install fileinfo package and adding this requirements you added external dependencies that not everybody can fulfill.
I and others installed php as required, and now it is not enough...

PS: I found a similar problem with gzdecode... in this case package php5-zlib is installed but this function is not present..

**Comment by fperillo** [comment]
I understand, and I really don't know Joomla developers guidelines and, believe me, I don't want to be rude, but after a hot phone call with one user... 
according to
https://downloads.joomla.org/technical-requirements
there is no need to install fileinfo package and adding this requirements you added external dependencies that not everybody can fulfill.
I and others installed php as required, and now it is not enough...

PS: I found a similar problem with gzdecode... in this case package php5-zlib is installed but this function is not present..

**Comment by brianteeman** [comment]
@fperillo fileinfo - This extension is enabled by default as of PHP 5.3.0. (http://php.net/manual/en/fileinfo.installation.php) 

Joomla cannot be expected to tell you every php extension that needs to be enabled especially if it is enabled by default.

**Comment by EndeavorC** [comment]
My hosting company verified that both are installed but I am still unable to upload image files with 3.7.1 or the updated files. If I use the 3.7.0 files I am able to upload image files. I have not tried PDF files as we use another option for document.

This is becoming very frustrating, the past 2 updates have had serious bugs and/or changes that have greatly impacted the volunteers that use the site.



________________________________

**Comment by brianteeman** [comment]
We are all volunteers as well

I see from the system info report that you posted before that you are using the ark editor on your site. If I understand your reports correctly you are able to upload images through the administrator but your editors can not at the front end. Are the editors using the ARK editor or the default tinymce editor? If they are using ARK can you try with the TinyMCe editor. I am not passing any blame here i am just trying to gather all the facts so that we can try to help you

**Comment by EndeavorC** [comment]
I have tried TinyMCE and ArkEditor from the front end with the same result, for some reason it does not work on the front end. I am not laying any blame but trying to provide what little troubleshooting or helpful information I may be able to provide.



________________________________

**Comment by jen4web** [comment]
I have been uploading files through the Media Manager.

I reported this to Rochen, who confirmed that finfo_open and mime_content_type were installed, available, and working correctly.

Their response was this: "Well there seems to be compatibility issue with mime_content_type on Joomla version and PHP v7.0, If I switch the PHP version to v5.6 it is working fine.

It appears the patch provided isn't working on 3.7.1 and 3.7.2, due to mime_content_type. Thanks!"

Previously I was running PHP 7.0.19.

**Comment by brianteeman** [comment]
Thanks for the info about php 7.0 - that will hopefully help track it down

But as this issue is closed it wont be seen by most people so can I please ask you to open a new issue with this information

**Comment by gwsdesk** [comment]
I am sorry but I cannot replicate this on our servers (PHP 7.1.14). Neither in default media manager nor in /TinyMCEJCE. I can upload pdf's either from frontend or admin without problems and if pdf's disabled (in media options) I do get the proper message (extension not supported)

**However:** The pdf's show in the file system with FTP (after upload in the media manager _but_ **they do not show after upload in the media manager** in the administrator (they are uploaded as stated since they show in the file manager with ftp) That only happens when uploaded in this case with Tiny/JCE but they show when uploaded in the admin panel in the media manager

**Comment by zero-24** [comment]
Thanks. The PDFs are not supported in the frontend image form. (different issue).

As stated above and here https://github.com/joomla/joomla-cms/issues/16238 it looks like a server / hosting issue.

Here you can find a PR with a better message incase your hosting did disable both checkers:
https://github.com/joomla/joomla-cms/pull/16246

I'm going to lock this issue now. If there are still issues please see #16238 or open a new issue. Thanks.

**Pull request merged**